Whisper.net/Ggml/WhisperGgmlDownloader.cs

```diff
@@ -1,27 +1,53 @@
 // Licensed under the MIT license: https://opensource.org/licenses/MIT
 
+using System.IO.Compression;
+
 namespace Whisper.net.Ggml;
 
 public static class WhisperGgmlDownloader
 {
     private static readonly Lazy<HttpClient> httpClient = new(() => new HttpClient() { Timeout = Timeout.InfiniteTimeSpan });
 
-    public static async Task<Stream> GetGgmlModelAsync(GgmlType type, CancellationToken cancellationToken = default)
+    /// <summary>
+    /// Gets the download stream for the model
+    /// </summary>
+    /// <param name="type">The type of the model which needs to be downloaded.</param>
+    /// <param name="quantization">The quantization of the model.</param>
+    /// <param name="cancellationToken">A cancellation token used to cancell the request to huggingface.</param>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    public static async Task<Stream> GetGgmlModelAsync(GgmlType type, QuantizationType quantization = QuantizationType.NoQuantization, CancellationToken cancellationToken = default)
     {
-        var url = type switch
-        {
-            GgmlType.Tiny => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-tiny.bin",
-            GgmlType.TinyEn => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-tiny.en.bin",
-            GgmlType.Base => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-base.bin",
-            GgmlType.BaseEn => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-base.en.bin",
-            GgmlType.Small => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-small.bin",
-            GgmlType.SmallEn => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-small.en.bin",
-            GgmlType.Medium => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-medium.bin",
-            GgmlType.MediumEn => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-medium.en.bin",
-            GgmlType.LargeV1 => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-large-v1.bin",
-            GgmlType.Large => "https://huggingface.co/datasets/ggerganov/whisper.cpp/resolve/main/ggml-large.bin",
-            _ => throw new ArgumentOutOfRangeException(nameof(type), type, null)
-        };
+        var subdirectory = GetQuantizationSubdirectory(quantization);
+        var modelName = GetModelName(type);
+
+        var url = $"https://huggingface.co/sandrohanea/whisper.net/resolve/v1/{subdirectory}/{modelName}.bin";
+
+        var request = new HttpRequestMessage(HttpMethod.Get, url);
+        var response = await httpClient.Value.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+        response.EnsureSuccessStatusCode();
+
+#if IOS || MACCATALYST || TVOS || ANDROID || MACOS
```

Comment:
You could filp this to `NETSTANDARD` instead

```csharp
#if NETSTANDARD
        return await response.Content.ReadAsStreamAsync();
#else
        return await response.Content.ReadAsStreamAsync(cancellationToken);
#endif
```

so it would automatically catch net6+ platforms with the cancellationToken.

Reply:
Thanks for the suggestion, simplified it as suggested.